### PR TITLE
Redefine alerting property

### PR DESCRIPTION
**Pull request by SwannCroiset**
The alerting property can be one of 'disabled', 'enabled' or
'enabled_with_notification'

### CODE REVIEW

File: heka/files/toml/filter/afd_alarm.toml [hunk 1]
```diff
@@ -11,4 +11,17 @@ afd_file = "lma_{{ alarm_name|replace('-', '_') }}"
 afd_name = "{{ alarm_name }}"
 hostname = "{{ grains.host }}"
 dimensions = '{{ salt['heka_alarming.dimensions'](alarm)|json }}'
-activate_alerting = {{ alarm.alerting|default(True)|lower }}
+
+{%- set alerting = alarm.get('alerting', 'enabled') %}
+
+{%- if alerting != 'disabled' and alerting != False %}
```
Comment (elemoine) [inline heka/files/toml/filter/afd_alarm.toml]: s/!=/is not/ ?

File: heka/files/toml/filter/afd_alarm.toml [hunk 2]
```diff
@@ -11,4 +11,17 @@ afd_file = "lma_{{ alarm_name|replace('-', '_') }}"
...
+activate_alerting = true
+{%- else %}
+activate_alerting = false
+{%- endif %}
+
+{%- if alerting == 'enabled_with_notification' %}
+enable_notification = true
+{%- else %}
+enable_notification = false
+{% endif %}
```
Comment (elemoine) [inline heka/files/toml/filter/afd_alarm.toml]: dash missing

**Comment by elemoine** [comment]
Shouldn't the same be done in the gse_alarm_cluster.toml file? https://github.com/tcpcloud/salt-formula-heka/blob/stacklight/heka/files/toml/filter/gse_alarm_cluster.toml

**Comment by elemoine** [comment]
Also, I don't think the Lua code handles `enable_notification` right now.

**Comment by SwannCroiset** [comment]
> s/!=/is not/

```
 Comment: Unable to manage file: Jinja syntax error: expected token 'name', got 'string'; line 17
          ---
          [...]
          hostname = "{{ grains.host }}"
          dimensions = '{{ salt['heka_alarming.dimensions'](alarm)|json }}'
          {%- set alerting = alarm.get('alerting', 'enabled') %}      
          {%- if alerting is not 'disabled' and alerting is not False %}    <======================
```

**Comment by SwannCroiset** [comment]
> Shouldn't the same be done in the gse_alarm_cluster.toml file? https://github.com/tcpcloud/salt-formula-heka/blob/stacklight/heka/files/toml/filter/gse_alarm_cluster.toml

yep

**Comment by SwannCroiset** [comment]
> Also, I don't think the Lua code handles enable_notification right now.

you are right .. it should be handled for smtp notification (email sent by heka)

File: heka/files/toml/filter/afd_alarm.toml [hunk 3]
```diff
@@ -11,4 +11,17 @@ afd_file = "lma_{{ alarm_name|replace('-', '_') }}"
 afd_name = "{{ alarm_name }}"
 hostname = "{{ grains.host }}"
 dimensions = '{{ salt['heka_alarming.dimensions'](alarm)|json }}'
-activate_alerting = {{ alarm.alerting|default(True)|lower }}
+
+{%- set alerting = alarm.get('alerting', 'enabled') %}
```
Comment (simonpasquier) [inline heka/files/toml/filter/afd_alarm.toml:14]: would be better to do it in _modules/heka_alarming.py

**Review by simonpasquier — CHANGES_REQUESTED** [review]

File: heka/files/toml/filter/afd_alarm.toml (hunk 3 above)
Comment (SwannCroiset) [reply heka/files/toml/filter/afd_alarm.toml:14]: > would be better to do it in _modules/heka_alarming.py

done

File: heka/files/toml/filter/afd_alarm.toml [hunk 4]
```diff
@@ -11,4 +11,5 @@ afd_file = "lma_{{ alarm_name|replace('-', '_') }}"
 afd_name = "{{ alarm_name }}"
 hostname = "{{ grains.host }}"
 dimensions = '{{ salt['heka_alarming.dimensions'](alarm)|json }}'
-activate_alerting = {{ alarm.alerting|default(True)|lower }}
+activate_alerting = {{ salt['heka_alarming.alarm_activate_alerting'](alarm.get('alerting', 'enabled')) }}
+enable_notification = {{ salt['heka_alarming.alarm_enable_notification'](alarm.get('alerting', 'disabled')) }}
```
Comment (elemoine) [inline heka/files/toml/filter/afd_alarm.toml]: I'd rewrite this as follows:

```
{%- set alerting = alarm.get('alerting', 'enabled') %}
activate_alerting = {{ salt['heka_alarming.alarm_activate_alerting'](alerting) }}
enable_notification = {{ salt['heka_alarming.alarm_enable_notification'](alerting) }}
```

File: heka/files/toml/filter/gse_alarm_cluster.toml [hunk 5]
```diff
@@ -19,3 +19,5 @@ max_inject = {{ alarm_cluster.max_inject }}
 {%- if alarm_cluster.warm_up_period is defined %}
 warm_up_period = {{ alarm_cluster.warm_up_period }}
 {%- endif %}
+activate_alerting = {{ salt['heka_alarming.alarm_activate_alerting'](alarm_cluster.get('alerting', 'enabled')) }}
+enable_notification = {{ salt['heka_alarming.alarm_enable_notification'](alarm_cluster.get('alerting', 'enabled_with_notification')) }}
```
Comment (elemoine) [inline heka/files/toml/filter/gse_alarm_cluster.toml]: Ditto.

**Review by elemoine — CHANGES_REQUESTED** [review]

File: heka/files/toml/filter/afd_alarm.toml (hunk 4 above)
Comment (SwannCroiset) [reply heka/files/toml/filter/afd_alarm.toml]: @elemoine done

**Review by elemoine — APPROVED** [review]

File: heka/files/toml/filter/afd_alarm.toml (hunk 3 above)
Comment (SwannCroiset) [reply heka/files/toml/filter/afd_alarm.toml:14]: @simonpasquier done

**Pull request merged**